frontend/packages/topology/src/elements/BaseElement.ts

```diff
@@ -127,7 +129,18 @@ export default abstract class BaseElement<E extends ElementModel = ElementModel,
   }
 
   isVisible(): boolean {
-    return this.visible;
+    let { visible } = this;
+    try {
+      let parent = this.getParent();
+
+      while (visible && parent) {
+        visible = parent.isVisible() && (!isNode(parent) || !(parent as Node).isCollapsed());
+        parent = parent.getParent();
+      }
+      // eslint-disable-next-line no-empty
+    } catch (e) {}
+
+    return visible;
```

Comment:
You shouldn't need a loop because each call to `parent.isVisible()` will move up.
Use `this.parent` to avoid the try / catch.
```
    return (
      this.visible &&
      (!this.parent ||
        (this.parent.isVisible() && (!isNode(this.parent) || !this.parent.isCollapsed())))
    );
```